src/workerd/util/sentry.h

```diff
@@ -48,39 +48,44 @@ inline kj::StringPtr maybeOmitColoFromSentry(uint32_t coloId) {
   do {                                                                         \
     static bool logOnce KJ_UNUSED = [&]() {                                    \
-      KJ_LOG(WARNING, __VA_ARGS__);                                            \
+      KJ_LOG(severity, __VA_ARGS__);                                           \
       return true;                                                             \
     }();                                                                       \
   } while (0)
 
+// Log this to Sentry once ever per process. Typically will be better to use LOG_PERIODICALLY.
+#define LOG_WARNING_ONCE(...)                                                  \
+  LOG_ONCE(WARNING, __VA_ARGS__);
+
 // Log this to Sentry once ever per process. Typically will be better to use LOG_ERROR_PERIODICALLY.
 #define LOG_ERROR_ONCE(...)                                                    \
-  do {                                                                         \
-    static bool logOnce KJ_UNUSED = [&]() {                                    \
-      KJ_LOG(ERROR, __VA_ARGS__);                                              \
-      return true;                                                             \
-    }();                                                                       \
-  } while (0)
+  LOG_ONCE(ERROR, __VA_ARGS__);
 
-// Slightly more expensive than LOG_ERROR_ONCE. Avoid putting into a hot path (e.g. within a loop)
+// Slightly more expensive than LOG_ONCE. Avoid putting into a hot path (e.g. within a loop)
 // where an overhead of ~hundreds of nanoseconds per evaluation to retrieve the current time would
 // be prohibitive.
-#define LOG_ERROR_PERIODICALLY(...)                                            \
+#define LOG_PERIODICALLY(severity, ...)                                        \
   do {                                                                         \
     static kj::TimePoint KJ_UNIQUE_NAME(lastLogged) =                          \
         kj::origin<kj::TimePoint>() - 1 * kj::HOURS;                           \
     const auto now = kj::systemCoarseMonotonicClock().now();                   \
     const auto elapsed = now - KJ_UNIQUE_NAME(lastLogged);                     \
     if (KJ_UNLIKELY(elapsed >= 1 * kj::HOURS)) {                               \
       KJ_UNIQUE_NAME(lastLogged) = now;                                        \
-      KJ_LOG(ERROR, __VA_ARGS__);                                              \
+      KJ_LOG(severity, __VA_ARGS__);                                           \
     }                                                                          \
   } while (0)
 
+// Slightly more expensive than LOG_ERROR_ONCE. Avoid putting into a hot path (e.g. within a loop)
+// where an overhead of ~hundreds of nanoseconds per evaluation to retrieve the current time would
+// be prohibitive.
+#define LOG_ERROR_PERIODICALLY(...)                                            \
```

Comment:
For symmetry with LOG_ONCE, you should probably have LOG_WARNING_PERIODICALLY.

Reply:
:+1: added it. I initially left it out, because I think that people should just use LOG_PERIODICALLY directly, and think that the specific LOG_WARNING/LOG_ERROR variants are regrettable, but I'm not eager to go refactor them throughout the codebase.